exchanges/credentials.go

```diff
@@ -0,0 +1,378 @@
+func (a *API) SetSecret(secret string) {
+	if a.credentials == nil {
+		a.credentials = &Credentials{}
+	}
+	a.credentials.Secret = secret
+}
+
+// SetClientID sets new clientID for the default credentials
+func (a *API) SetClientID(clientID string) {
+	if a.credentials == nil {
+		a.credentials = &Credentials{}
+	}
+	a.credentials.ClientID = clientID
+}
+
+// SetPEMKey sets pem key for the default credentials
+func (a *API) SetPEMKey(pem string) {
+	if a.credentials == nil {
+		a.credentials = &Credentials{}
+	}
+	a.credentials.PEMKey = pem
+}
+
+// SetSubaccount sets sub account for the default credentials
+func (a *API) SetSubAccount(sub string) {
+	if a.credentials == nil {
+		a.credentials = &Credentials{}
+	}
+	a.credentials.SubAccount = sub
+}
+
+// CheckCredentials checks to see if the required fields have been set before
+// sending an authenticated API request
+func (b *Base) CheckCredentials(creds *Credentials, isContext bool) error {
+	if b.SkipAuthCheck {
+		return nil
+	}
+
+	// Individual package usage, allow request if API credentials are valid a
+	// and without needing to set AuthenticatedSupport to true
+	if !b.LoadedByConfig {
+		return b.ValidateAPICredentials(creds)
+	}
+
+	// Bot usage, AuthenticatedSupport can be disabled by user if desired, so
+	// don't allow authenticated requests. Context credentials set will override
+	// default credentials and supported checks.
+	if !b.API.AuthenticatedSupport && !b.API.AuthenticatedWebsocketSupport && !isContext {
+		return fmt.Errorf("%s %w", b.Name, ErrAuthenticationSupportNotEnabled)
+	}
+
+	// Check to see if the user has enabled AuthenticatedSupport, but has
+	// invalid API credentials set and loaded by config
+	return b.ValidateAPICredentials(creds)
+}
+
+// AreCredentialsValid returns if the supplied credentials are valid.
+func (b *Base) AreCredentialsValid(ctx context.Context) bool {
+	creds, err := b.GetCredentials(ctx)
+	return err == nil && b.ValidateAPICredentials(creds) == nil
+}
+
+// GetDefaultCredentials returns the exchange.Base api credentials loaded by
+// config.json
+func (b *Base) GetDefaultCredentials() *Credentials {
+	return b.API.credentials
+}
+
+// GetCredentials checks and validates current credentials, context credentials
+// override default credentials, if no credentials found, will return an error.
+func (b *Base) GetCredentials(ctx context.Context) (*Credentials, error) {
+	value := ctx.Value(contextCrendentialsFlag)
+	if value != nil {
+		ctxCredStore, ok := value.(*contextCredentialsStore)
+		if !ok {
+			return nil, errContextCredentialsFailure
+		}
+
+		creds := ctxCredStore.Get()
+		if err := b.CheckCredentials(creds, true); err != nil {
+			return nil, fmt.Errorf("context credentials issue: %w", err)
+		}
+		return creds, nil
+	}
+	return b.API.credentials, b.CheckCredentials(b.API.credentials, false)
+}
+
+// ValidateAPICredentials validates the exchanges API credentials
+func (b *Base) ValidateAPICredentials(creds *Credentials) error {
```

Comment:
Not 100% sure where to put this, but what about having the subaccount field being applied on its own such as the following:
```
go run . --apisubaccount="imARealSubAccount" --timeout=5m getcollateral --exchange="ftx" --asset="futures" --i
```
and having it only override the subaccount field and the rest remain defaults? The latest change wrt the command above can no longer request a specific subaccount without also providing api keys per request